src/react/components/common/tagInput/tagInput.tsx

```diff
@@ -611,17 +633,27 @@ export class TagInput extends React.Component<ITagInputProps, ITagInputState> {
         return menuItems;
     }
 
+    private isTypeCompatibleWithTag = (tag, type) => {
+        // If free tag we can assign any type
+        if (tag && tag.documentCount <= 0) {
+            return true;
+        }
+        const tagType = this.getTagCategory(tag.type);
+        const menuItemType = this.getTagCategory(type);
+        return tagType === menuItemType;
+    }
+
     private getTypeSubMenuItems = (): IContextualMenuItem[] => {
         const tag = this.state.selectedTag;
         const types = Object.values(FieldType);
-
         return types.map((type) => {
             return {
                 key: type,
                 text: type,
-                canCheck: true,
+                canCheck: this.isTypeCompatibleWithTag(tag, type),
```

Comment:
Can we just call the method once and save the result to a const and use it in const?
const isCompatible = this.isTypeCompatibleWithTag(tag, type);
...
canCheck: isCompatible,
...
disabled: !isCompatible,